csm/conf/setup.py

```diff
@@ -604,6 +604,23 @@ def _resolve_faulty_node_alerts(self, node_id):
         except Exception as ex:
             raise CsmSetupError(f"Refresh Context: Resolving of alerts failed. {ex}")
 
+    @staticmethod
+    def _is_vm():
+        setup_info = Setup.get_data_from_provisioner_cli(const.GET_SETUP_INFO)
+        return setup_info[const.SERVER_TYPE] == 'virtual'
+
+    @staticmethod
+    def _get_unsupported_features_from_schema():
+        schema = Json(const.UNSUPPORTED_FEATURE_SCHEMA).load()
+        all_unsupported_features = {
+            e[const.NAME]: e[const.UNSUPPORTED_FEATURES] for e in
+            schema[const.SETUP_TYPES]
+        }
+        setup_info = Setup.get_data_from_provisioner_cli(const.GET_SETUP_INFO)
+        storage_type = setup_info.get(const.STORAGE_TYPE)
```

Comment:
Already doing this operation in `_is_vm()`. Please avoid calling it twice.

Reply:
I will memoize the results so the costs of multiple calls becomes neligible.

Reply:
Done.

Reply:
We can Simply have Class variable that stores unsupported feature and setup info in memory to avoid multiple calls, so that it can be used whereaver required trough out the program execution.

Reply:
The memoization trick is already in place, please have a look.  Python is not a strictly OOP language and there is no need to pollute the class with helper attributes for such purpose.  This is bad style (even in languages like Java I would say).